R/MetaAnalysis.R

```diff
@@ -0,0 +1,424 @@
+                              method = "bayesian",
+                              resultsZipPattern = "^Results_.*\\.zip",
+                              addTraditional = TRUE) {
+  
+  if (!file.exists(maExportFolder)) {
+    dir.create(maExportFolder, recursive = TRUE)
+  }
+  
+  ParallelLogger::addDefaultFileLogger(file.path(maExportFolder, "metaAnalysisLog.txt"))
+  on.exit(ParallelLogger::unregisterLogger("DEFAULT_FILE_LOGGER", silent = TRUE))
+  
+  if (!file.exists(allDbsFolder)) {
+    stop(sprintf("The allDbsFolder path does not exist:\n\t", allDbsFolder))
+  }
+  
+  if (!(method %in% c("bayesian", "DL"))) {
+    stop("Accepted method arguments are \"bayesian\" or \"DL\"")
+  }
+  
+  resultsSets <- list.files(path = allDbsFolder, pattern = resultsZipPattern)
+  ParallelLogger::logInfo(sprintf("Found %d zip files matching pattern %s for synthesizing", length(resultsSets), resultsZipPattern))
+  if(length(resultsSets) == 0) {
+    stop(sprintf("No results found matching pattern %s in directory %s", resultsZipPattern, allDbsFolder))
+  } else if(length(resultsSets) == 1) {
+    stop(sprintf("Only single result set found matching pattern %s in directory %s", resultsZipPattern, allDbsFolder))
+  }
+  
+  mainResults <- lapply(resultsSets, loadDatabaseResults, allDbsFolder = allDbsFolder)
+  mainResults <- do.call(rbind, mainResults)
+  mainResults <- split(mainResults, paste(mainResults$targetId, mainResults$comparatorId, mainResults$analysisId))
+  
+  if(method == "bayesian") {
+    profiles <- lapply(resultsSets, loadLikelihoodProfiles, allDbsFolder = allDbsFolder)
+    profiles <- do.call(rbind, profiles)
+    profiles <- split(profiles, paste(profiles$targetId, profiles$comparatorId, profiles$analysisId))
+    
+    groups <- lapply(names(mainResults), function(name) list(mainResults = mainResults[[name]], profiles = profiles[[name]]))
+    
+    rm(mainResults)
+    rm(profiles)
+    
+  } else {
+    groups <- lapply(names(mainResults), function(name) list(mainResults = mainResults[[name]]))
+  }
+  
+  ParallelLogger::logInfo("Performing cross-database evidence synthesis")
+  cluster <- ParallelLogger::makeCluster(min(maxCores, 10))
+  indices <- 1:length(groups)
+  results <- ParallelLogger::clusterApply(cluster, indices, SkeletonComparativeEffectStudy:::computeGroupMetaAnalysis, groups, method, addTraditional)
+  ParallelLogger::stopCluster(cluster)
+  
+  results <- do.call(rbind, results)
+  results$trueEffectSize <- NULL
+  
+  tempFolder <- tempfile()
+  dir.create(tempFolder, recursive = TRUE)
+  on.exit(unlink(tempFolder, recursive = TRUE, force = TRUE))
+  
+  colnames(results) <- SqlRender::camelCaseToSnakeCase(colnames(results))
+  fileName <-  file.path(tempFolder, "cohort_method_result.csv")
+  write.csv(results, fileName, row.names = FALSE)
+  
+  ParallelLogger::logInfo("Creating harmonized database table")
+  databases <- lapply(resultsSets, loadDatabase, allDbsFolder = allDbsFolder)
```

Comment:
I don't think `databases` is used anywhere? I also can't think of a reason why you'd need this here. The database.csv files are merged when the data from all databases (and meta-analysis) are merged in the premerge file.

Reply:
Good catch - this was remnants of computing min/max observation periods across result sites.  I did remove for now, but I left the ```loadDatabase``` function for posterity (let me know if you think this should be removed, too).